.dotnet/src/Custom/Chat/ChatTokenUsage.cs

```diff
@@ -7,12 +7,28 @@ namespace OpenAI.Chat;
 public partial class ChatTokenUsage
 {
     // CUSTOM: Renamed.
-    /// <summary> Number of tokens in the generated completion. </summary>
+    /// <summary>
+    /// The combined number of output tokens consumed by the model.
+    /// </summary>
+    /// <remarks>
+    /// When using a model that supports <see cref="ReasoningTokens"/> such as <c>o1-mini</c>, this value represents
+    /// the sum of those reasoning tokens and conventional, displayed output tokens.
+    /// </remarks>
     [CodeGenMember("CompletionTokens")]
     public int OutputTokens { get; }
 
-    // CUSOTM: Renamed.
-    /// <summary> Number of tokens in the prompt. </summary>
+    // CUSTOM: Renamed.
+    /// <summary>
+    /// The number of tokens in the request message input, spanning all message content items.
+    /// </summary>
     [CodeGenMember("PromptTokens")]
     public int InputTokens { get; }
+
+    /// <summary>
+    /// The count of reasoning tokens used. Currently applicable to `o1` series models.
+    /// </summary>
+    public int ReasoningTokens => _internalCompletionTokenDetails?.ReasoningTokens ?? 0;
```

Comment:
I'm fine with separating. Let's see -- the JSON is:

```json
"usage": {
  "prompt_tokens": 123,
  "completion_tokens": 456,
  "total_tokens": 789,
  "completion_tokens_details": {
    "reasoning_tokens": 123
  }
}
```

Since we align "prompt" with "input" and "completion" with "output" (which I think is very useful), how about we snap to something like:

```csharp
public class ChatTokenUsage {
  public int InputTokenCount { get; }
  public int OutputTokenCount { get; }
  public ChatOutputTokenUsageDetails OutputTokenDetails { get; }
}
public class ChatOutputTokenUsageDetails {
  public int ReasoningTokenCount { get; }
}
```

Reply:
This looks good to me!